File: app/modules/peer_fallback_targets/schemas.py
from __future__ import annotations

from datetime import datetime

from pydantic import Field

from app.modules.shared.schemas import DashboardModel


class PeerFallbackTargetResponse(DashboardModel):
    id: str
    base_url: str
    enabled: bool
    created_at: datetime
    updated_at: datetime


class PeerFallbackTargetsResponse(DashboardModel):
    targets: list[PeerFallbackTargetResponse] = Field(default_factory=list)


class PeerFallbackTargetCreateRequest(DashboardModel):
    base_url: str
    enabled: bool = True


class PeerFallbackTargetUpdateRequest(DashboardModel):
    base_url: str | None = None
    enabled: bool | None = None


class PeerFallbackTargetDeleteResponse(DashboardModel):
    status: str

File: frontend/src/features/peer-fallback-targets/schemas.ts
import { z } from "zod";

export const PeerFallbackTargetSchema = z.object({
  id: z.string().min(1),
  baseUrl: z.string().url(),
  enabled: z.boolean(),
  createdAt: z.string().datetime({ offset: true }),
  updatedAt: z.string().datetime({ offset: true }),
});

export const PeerFallbackTargetsResponseSchema = z.object({
  targets: z.array(PeerFallbackTargetSchema).default([]),
});

export const PeerFallbackTargetCreateRequestSchema = z.object({
  baseUrl: z.string().url(),
  enabled: z.boolean().optional(),
});

export const PeerFallbackTargetUpdateRequestSchema = z.object({
  baseUrl: z.string().url().optional(),
  enabled: z.boolean().optional(),
});

export const PeerFallbackTargetDeleteResponseSchema = z.object({
  status: z.string().min(1),
});

export type PeerFallbackTarget = z.infer<typeof PeerFallbackTargetSchema>;
export type PeerFallbackTargetsResponse = z.infer<typeof PeerFallbackTargetsResponseSchema>;
export type PeerFallbackTargetCreateRequest = z.infer<typeof PeerFallbackTargetCreateRequestSchema>;
export type PeerFallbackTargetUpdateRequest = z.infer<typeof PeerFallbackTargetUpdateRequestSchema>;
export type PeerFallbackTargetDeleteResponse = z.infer<typeof PeerFallbackTargetDeleteResponseSchema>;
